x-pack/plugins/fleet/server/services/artifacts/artifacts.ts

```diff
@@ -57,7 +58,7 @@ export const createArtifact = async (
   esClient: ElasticsearchClient,
   artifact: NewArtifact
 ): Promise<Artifact> => {
-  const id = uuid.v4();
+  const id = `${artifact.packageName}:${artifact.identifier}-${artifact.decodedSha256}`;
```

Comment:
Perhaps this could be pulled into a function which takes an artifact and returns an id, but not a blocker.

Reply:
Yeah, I thought about that as well and for whatever reason I did not do it. I'll change now. 😄 thanks